sunpy/map/maputils.py

```diff
@@ -375,3 +375,25 @@ def on_disk_bounding_coordinates(smap):
     return SkyCoord([np.nanmin(tx), np.nanmax(tx)] * u.arcsec,
                     [np.nanmin(ty), np.nanmax(ty)] * u.arcsec,
                     frame=smap.coordinate_frame)
+
+def contains_coordinate(smap, coordinates):
+    """
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    coordinate : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate. The coordinate frame must be
+        `~sunpy.coordinates.Helioprojective`.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` If coordinates are within the coordinate
+        bound of the smap and Returns `False` If coordinates are 
+        not within the coordinate bound of the smap.
+    """
+    _verify_coordinate_helioprojective(smap.coordinate_frame)
+
+    return coordinates in all_coordinates_from_map(smap)
```

Comment:
This implementation would not give the correct answer if input coordinate did not correspond exactly with one of the pixels in the map, plus it's very inefficient.  You should convert the input coordinate to pixel space and check if it falls within the pixel bounds.

Reply:
i am new to OSC and this my first time trying it . "You should convert the input coordinate to pixel space and check if it falls within the pixel bounds" . please tell me how to convert coordinate to pixel space

Reply:
You can use the `world_to_pixel` method: https://docs.sunpy.org/en/stable/api/sunpy.map.GenericMap.html?highlight=world_to_pixel#sunpy.map.GenericMap.world_to_pixel. Once you have the pixels, you can then check that they are within the limits of the map.

Reply:
Can we use this method? -> with the help of ```world_to_pixel()``` method, Find the pixel of coordinates and then find out the maximum and minimum values of edges of map by ```map_edges()``` method and check whether the coordinate pixels are within the max and min values. 
And I am having a doubt that, if the coordinate is out of bounds how will it give pixel to the method ```world_to_pixel()```, as it is not there in the map?

Reply:
Shouldn't even need to use `map_edges()`, can just compare the pixel values to the size of the data array. In general a coordinate > pixel transformation doesn't depend on the pixel boundaries, so there's no need to worry about out of bounds coordinates.